website/content/docs/custom-widgets.md

```diff
@@ -118,29 +118,69 @@ CMS.registerEditorComponent(definition)
 <script>
 CMS.registerEditorComponent({
   // Internal id of the component
-  id: "youtube",
+  id: "collapsible-note",
   // Visible label
-  label: "Youtube",
+  label: "Collapsible Note",
   // Fields the user need to fill out when adding an instance of the component
-  fields: [{name: 'id', label: 'Youtube Video ID', widget: 'string'}],
-  // Pattern to identify a block as being an instance of this component
-  pattern: /^youtube (\S+)$/,
-  // Function to extract data elements from the regexp match
+  fields: [
+    {
+      name: 'summary',
+      label: 'Summary',
+      widget: 'string'
+    },
+    {
+      name: 'details',
+      label: 'Details',
+      widget: 'markdown'
+    }
+  ],
+  // Regex pattern used to search for instances of this block in the markdown document.
+  // Patterns are run in a multline environment (against the entire markdown document),
+  // and so generally should make use of the multiline flag (`m`). If you need to capture
+  // newlines in your capturing groups, you can either use something like
+  // `([\S\s]*)`, or you can additionally enable the "dot all" flag (`s`),
+  // which will cause `(.*)` to match newlines as well.
+  //
+  // Additionally, it's recommended that you use non-greedy capturing groups (e.g.
+  // `(.*?)` vs `(.*)`), especially if matching against newline characters.
+  pattern: /^<details>\s*?<summary>(.*?)<\/summary>\n\n(.*?)\n<\/details>$/ms,
```

Comment:
@stefanprobst what do you think? I beefed up the example itself so that it could make more use of the features being described in the documentation. Open to edits!